### PR TITLE
feat: update version to 6.2.0

### PR DESCRIPTION
**Pull request by dalisoft**
fix: fixed version updated
feat: set `auto_updates` to `true`
feat: added `livecheck` block

### CODE REVIEW

**Comment by apjanke** [comment]
Octave.app 6.2.0 is not out yet, so can't merge this yet. Planning on getting a 6.2.0 build out this weekend, though!

**Comment by dalisoft** [comment]
Thanks for reply. 
it is fake then? https://github.com/octave-app/octave-app/releases/tag/v6.2.0

**Comment by apjanke** [comment]
Oh, wait, 6.2.0 _is_ out; I completely forgot we did that!

Okay, I think I can merge this later today.

**Comment by apjanke** [comment]
I don't think `auto_updates true` should be here. `auto_updates true` means that the application updates itself (without `brew`'s involvement), and `brew cask outdated` and `brew cask upgrade` should ignore it. Octave.app doesn't do that; users will probably want the `brew cask update` mechanism to keep working.

Could you leave `auto_updates` as `false`, and we can get this merged? Or am I missing something here?

File: Casks/octave-app.rb [hunk 1]
```diff
@@ -1,14 +1,20 @@
...
+  livecheck do
+    url "https://github.com/octave-app/octave-app/releases"
+    strategy :page_match
+    regex(%r{href=.*?/Octave[._-]?(\d+(?:\.\d+)+)\.dmg}i)
+  end
+
+  depends_on macos: '>= :mojave'
 
-  app 'Octave-6.1.0.app'
+  app 'Octave-#{version}.app'
```
Comment (apjanke) [inline Casks/octave-app.rb]: I'm not sure this works. Does `#{...}` variable interpolation work inside single quotes? When I do a test run against this PR, I get the following error:

```
$ brew reinstall octave-app
==> Downloading https://github.com/octave-app/octave-app/releases/download/v6.2.0/Octave-6.2.0.dmg
==> Downloading from https://github-releases.githubusercontent.com/137706143/c33a9980-9e4d-11eb-9468-ed672fca39ef?X-Amz-Algorithm=
######################################################################## 100.0%
==> Uninstalling Cask octave-app
==> Purging files for version 6.1.0 of Cask octave-app
==> Installing Cask octave-app
==> Purging files for version 6.2.0 of Cask octave-app
Error: It seems the App source '/usr/local/Caskroom/octave-app/6.2.0/Octave-#{version}.app' is not there.
```

**Review by apjanke — CHANGES_REQUESTED** [review]

File: Casks/octave-app.rb (hunk 1 above)
Comment (dalisoft) [reply Casks/octave-app.rb]: I fixed it. Thank you for testing

**Review by apjanke — APPROVED** [review]

**Comment by apjanke** [comment]
Good to go. Merged. Thank you!

**Pull request merged**